### ISSUE TITLE
Generic class parameterized by TypeVar with default isn't specialized for attribute access

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#7445]
PEP 696 states:
# > Type parameter defaults should be bound by attribute access (including call and subscript).

That means the following code should not generate an error,  but it currently does.

```python
from typing import Generic, assert_type
from typing_extensions import TypeVar

DefaultIntT = TypeVar("DefaultIntT", default=int)

class Foo7(Generic[DefaultIntT]):
    attr: DefaultIntT

assert_type(Foo7.attr, int)
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.354, which I just published.
